core-nonfree/story-api-nonfree/src/main/java/edu/cmu/cs/dennisc/nebulous/GenericModelAdapter.java

```diff
@@ -86,15 +87,14 @@ protected void renderGeometry(RenderContext rc, GlrVisual.RenderType renderType)
   }
 
   @Override
-  public Point3 getIntersectionInSource(Point3 rv, Ray ray, AffineMatrix4x4 m, int subElement) {
-    Vector3 direction = Vector3.createNegation(m.translation);
-    direction.y = 0.0;
-    if (direction.calculateMagnitudeSquared() == 0.0) {
-      rv.setNaN();
+  public Point3 getIntersectionInSource(Ray ray, Matrix4x4 m, int subElement) {
```

Comment:
I do notice that we used to negate and now we don't? I'm assuming you thought about this one and this is correct

Reply:
You give me too much credit.
I will restore the negation.

This is only used during picking. It is invoked in the else blocks of `isPickFunctioningCorrectly` so it remains an edge case on an edge case.